### DISCUSSION TITLE
Formik is re-rendering every-time I click on the page, why?

### DISCUSSION THREAD

**Discussion by kewitz** [jaredpalmer/formik#2836] [Q&A]
For some weird reason, every time I click on the page that has Formik rendered, it will cause Formik to re-render.
Any clues why this is happening? I tried to set "validateOnBlur = false" but that did not work.

![deepin-screen-recorder_chromium_20201023150715](https://user-images.githubusercontent.com/2119706/97038697-931ad980-1541-11eb-9106-16530a0bccb1.gif)

**Comment by johnrom** [comment]
I can't get a good view of your hierarchy from this gif because it scrolls so quickly through the graph, but this problem is usually indicative of an element creating a new element on each render that cannot be determined by React to be the same (same key, etc). Usually this happens if you do something like:

```jsx
const getForm = () => <Formik />;

const MyPage = () => {
  const MyForm = getForm();

  return <div>{ MyForm }</div>;
}
```

It also happens frequently when using `array.map` and other iterative scenarios; those are generally indicated by a console warning like "Warning: Each component in an array must have a unique key".

**Reply by kewitz** [reply]
I don't think that's the case.
The React dev-tools explicitly show that Formik was triggering the render because "Hooks changed".
![image](https://user-images.githubusercontent.com/2119706/97054927-206b2780-155c-11eb-98a7-532526a10a45.png)
![image](https://user-images.githubusercontent.com/2119706/97053881-206a2800-155a-11eb-97e2-30db25d375b6.png)

**Reply by johnrom** [reply]
It's pretty hard to tell why hooks would have changed without being able to see the code. Does any of your code or the third party components you are using include click, focus or blur event listeners which could cause this?

If you look at the base Formik example, I don't see this rendering issue in formik@2.2.1.

https://codesandbox.io/s/formik-v2-tutorial-final-forked-5xwb8

**Reply by kewitz** [reply]
I'll try updating Formik and if that doesn't work I'll debug this further, thanks @johnrom!
